azpubsub/src/main/java/com/microsoft/log4j/DatedRollingFileAppender.java

```diff
@@ -103,7 +120,36 @@ public int getMaxBackupIndex() {
 
     public synchronized void setFile(String fileName, boolean append, boolean bufferedIO, int bufferSize) throws IOException {
         this.originalFileName = fileName;
-        super.setFile(this.originalFileName + sdf.format(new Date()), append, bufferedIO, bufferSize);
+        String name = this.originalFileName + sdf.format(new Date());
```

Comment:
private String getSerialNumber() throws IOException {
        // find the current/latest log file for the current date
        Path logFile = Paths.get(originalFileName);
        Pattern pattern = Pattern.compile("^" + logFile.getFileName() + sdf.format(new Date()) + "(.(?<sno>\\d+))?$");
        Optional<String> file = 
        		Files.find(
        				logFile.getParent(), 
        				1, 
		                (path, basicFileAttributes) -> {
		                	Matcher matcher = pattern.matcher(path.getFileName().toString());
		                	if (matcher.matches()) {
		                		return true;
		                	}
		                    return false;
		                })
	        		.map(p -> p.toFile().getName())
	        		.max(Comparator.comparing(String::valueOf));

        // extract the serial number if the current/latest log file exist
        int sno = 1;
        if (file.isPresent()) {
        	String filename = file.get();
        	Matcher matcher = pattern.matcher(filename);
        	if (matcher.matches()) {
        		if (matcher.group("sno") != null) {
        			sno = Integer.parseInt(matcher.group("sno"));
        			File f = Paths.get(logFile.getParent().toString(), filename).toFile();
        			// if the current/latest file reached the max file size, move it to next
        			if (f.length() >= 10) {
        				sno++;
        			}
        		}
        	}
        }

        return String.format("%04d", sno);
	}